crates/esthri/src/ops/sync.rs

```diff
@@ -445,10 +449,51 @@ where
 
     sync_tasks
         .buffer_unordered(task_count)
-        .try_collect()
+        .try_for_each_concurrent(task_count, |_| future::ready(Ok(())))
         .await?;
 
-    Ok(())
+    if delete {
+        sync_delete_local(s3, bucket, key, directory, filters, task_count).await
+    } else {
+        Ok(())
+    }
+}
+
+async fn sync_delete_local<T>(
+    s3: &T,
+    bucket: &str,
+    key: &str,
+    directory: &Path,
+    filters: &[GlobFilter],
+    task_count: usize,
+) -> Result<()>
+where
+    T: S3 + Send + Sync + Clone,
+{
+    let stream = flattened_object_listing(s3, bucket, key, directory, filters, false);
```

Comment:
This touches on one of the questions that came up in the PP session we had with @achubbic-snav which was how the glob filters should be handled with respect to the "delete" feature of sync. I believe here it is filtering what will be synced up ( include/exclude ) but also filtering what would be considered for deletion in the destination. Maybe I'm overthinking this.

```
include: *.txt
exclude: *
```

src/
  - asdf.txt
  - asdf2.txt
  - asdf3.mpg

dest/
  - asdf.txt
  - asdf2.txt
  - asdf3.mpg
  - asdf4.mpg

In this example due to the filters applied on both src and dest, I believe the `asdf4.mpg` would not be considered for deletion.

Reply:
We tested this with the aws cli and the filters applied to both upload/download and what was considered for deletion-- which coincidentally was the easiest thing to implement too.

Reply:
Right on. Interesting chicken and the egg problem.